### PR TITLE
Remove ignore requirement for tests needing root privileges

### PR DESCRIPTION
**Pull request by future-highway**
We've been marking tests that need sudo (root privileges) with ignore, so that we can run the tests in CI.

I noticed the Nix crate had some helpful macros, circumventing the need to use ignore. They aren't part of their public api, so I created a `test-helpers` crate for us and copied the 2 macros we needed to skip when root isn't available.

### CODE REVIEW

**Comment by future-highway** [comment]
Marked the PR as draft until I update it to run the root requiring tests in 035

File: Makefile [hunk 1]
```diff
@@ -357,6 +357,9 @@ alpine: ## Creat an "alpine" OCI bundle in target
 
 # CI Commands
 
+.PHONY: ci-test
+ci-test: pki config test
```
Comment (dmah42) [inline Makefile:361]: i'm not a huge fan (personal opinion) of having make targets just for convenience.  if the GHA should run `make pki config test` then let's do that so it's explicit what it's doing.  it keeps the local development workflow simple as `make test` should always pass and i don't need to think about what `make ci-test` is or why it's different.

File: auraed/src/cells/cell_service/cells/cell.rs [hunk 2]
```diff
@@ -244,11 +244,12 @@ impl Drop for Cell {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use test_helpers::*;
 
-    // Ignored: requires sudo, which we don't have in CI
-    #[ignore]
     #[test]
     fn test_cant_unfree() {
+        skip_if_not_root!("test_cant_unfree");
```
Comment (dmah42) [inline auraed/src/cells/cell_service/cells/cell.rs:251]: <3 this

**Review by dmah42 — APPROVED** [review]

**Comment by future-highway** [comment]
Closed in favor of #338

**Pull request closed without merging**